File: user-service/src/main/java/rs/edu/raf/dto/ChangePasswordWithCodeDTO.java
package rs.edu.raf.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;

public record ChangePasswordWithCodeDTO(
    @Pattern(regexp = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$", message = "Email address must be valid!")
    String email,
    @NotNull String password,
    @NotNull String code
) { }

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/dto/ChangePasswordWithCodeDTO.kt
package org.example.userservicekotlin.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern

data class ChangePasswordWithCodeDTO(
    @Pattern(regexp = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$", message = "Email address must be valid!")
    val email: String,
    @NotNull val password: String,
    @NotNull val code: String
)
